### DISCUSSION TITLE
Automatic workshop uploading via Github Actions

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2647] [Suggestions]
Since we don't actually make releases anymore I think it may be appropriate to have a workflow to automatically upload wiremod to the workshop.

Could be per-commit (probably don't want to do this), or maybe weekly/biweekly.

**Comment by vurvdev** [comment]
https://github.com/Meachamp/gmodws
https://github.com/Facepunch/gmad

https://github.com/marketplace/actions/gmod-steam-workshop

@Anticept 

I do want to get this done, need to decide how we're gonna go about it. Could use gmodws and try to clone/compile gmad manually through github actions but that seems it'd be a pain. Earu's github action [apparently doesn't work.](https://github.com/Earu/GSW-action/issues/5) (probably some issue with [their manual gma creator](https://github.com/Earu/GSW-action/blob/main/src/helpers/createGMA.ts)).

Could fork their workflow, make our own gma packer (seems pretty simple), lots of ways to go about this

**Comment by vurvdev** [comment]
Someone also made a gma packer that [gmpublisher](https://github.com/WilliamVenner/gmpublisher) uses: https://github.com/wyozi/steamws. [Relevant code here](https://github.com/wyozi/steamws/blob/9196321653a9a047b098a6c0c122e6a1775c158c/src/gma/write.rs#L17-L45)

**Comment by vurvdev** [comment]
Made an implementation using a gma packer I created in lua and [gmodws](https://github.com/Meachamp/gmodws). Testing on my addon, [but I encountered an issue with gmodws](https://github.com/Meachamp/gmodws/issues/14), so waiting on that. Could feasibly write my own library using the steamworks sdk but I'd prefer not to.

**Comment by vurvdev** [comment]
Got it working with another github action someone made for arma: https://github.com/Vurv78/WebAudio/blob/main/.github/workflows/deploy.yml

**Reply by thegrb93** [reply]
Can the action be made housed in our own repo? It's a big of a risk to trust outside actions with secrets.

**Reply by vurvdev** [reply]
Yeah I was gonna ping Anticept to make a fork of it in the @wiremod org when I got to making the actual PR

**Reply by vurvdev** [reply]
Actually I think I have permissions to do that myself

**Comment by Anticept** [comment]
Sorry I forgot all about this.

I have created the secrets for your use. WIRETEAM_WORKSHOP_USERNAME and WIRETEAM_WORKSHOP_PASSWORD
